grails-app/views/report/showTransactionReport.gsp

```diff
@@ -76,24 +76,28 @@
 								<label>
 									<warehouse:message code="report.location.label"/>
 								</label>
-								<g:selectLocation class="chzn-select-deselect filter"
-												  id="locationId"
-												  name="location.id"
-												  activityCode="${org.pih.warehouse.core.ActivityCode.MANAGE_INVENTORY}"
-												  noSelection="['':'']"
-												  maxChars="75"
-												  groupBy="locationType"
-												  value="${command?.location?.id}"/>
+								<g:selectLocation
+									class="chzn-select-deselect filter"
+									id="locationId"
+									name="location.id"
+									activityCode="${org.pih.warehouse.core.ActivityCode.MANAGE_INVENTORY}"
+									noSelection="['':'']"
+									maxChars="75"
+									groupBy="locationType"
+									value="${command?.location?.id}"
+								/>
 							</div>
 							<div class="filter-list-item">
 								<label><warehouse:message code="category.label"/></label>
 								<p>
-									<g:selectCategory id="category"
-													  class="chzn-select-deselect filter"
-													  data-placeholder="Select a category"
-													  name="category"
-													  noSelection="['':'']"
-													  value="${params?.category?.id?:command?.rootCategory?.id}"/>
+									<g:selectCategory
+										id="category"
+									    class="chzn-select-deselect filter"
+									    data-placeholder="${g.message(code: 'product.selectACategory.label', default: 'Select a category')}"
```

Comment:
In cases where the label / message has an article (a, the, etc) you can exclude the article in the code (even if you keep it in the translated value).
```
product.selectCategory.label=Select a category
```
also for this one the namespace should be category.
```
category.selectCategory.label=Select a category
```